probes/hasDangerousWorkflowScriptInjection/impl.go

```diff
@@ -53,25 +58,53 @@ func Run(raw *checker.RawResults) ([]finding.Finding, string, error) {
 	}
 
 	var findings []finding.Finding
+	var curr string
+	var workflow *actionlint.Workflow
+	var content []byte
+	var errs []*actionlint.Error
+	localPath := raw.Metadata.Metadata["localPath"]
 	for _, e := range r.Workflows {
 		e := e
-		if e.Type == checker.DangerousWorkflowScriptInjection {
-			f, err := finding.NewWith(fs, Probe,
-				fmt.Sprintf("script injection with untrusted input '%v'", e.File.Snippet),
-				nil, finding.OutcomeTrue)
+		if e.Type != checker.DangerousWorkflowScriptInjection {
+			continue
+		}
+		f, err := finding.NewWith(fs, Probe,
+			fmt.Sprintf("script injection with untrusted input '%v'", e.File.Snippet),
+			nil, finding.OutcomeTrue)
+		if err != nil {
+			return nil, Probe, fmt.Errorf("create finding: %w", err)
+		}
+		f = f.WithLocation(&finding.Location{
+			Path:      e.File.Path,
+			Type:      e.File.Type,
+			LineStart: &e.File.Offset,
+			Snippet:   &e.File.Snippet,
+		})
+		findings = append(findings, *f)
```

Comment:
> What's the issue? Would you prefer that we only append the finding once it's "finished"?

The issue is the probe returns a `[]finding`. As written, the code adds a copy of the finding to the slice when we deference the pointer `append(findings, *f)`, but then we go on to modify the original and the copy in slice doesn't change.

The fact that this works as written is accidental in my opinion, since both copies have a pointer to the same `finding.Remediation` object.

Reply:
Done, PTAL.